dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -146,23 +146,33 @@ class AppEngineCocoonService implements CocoonService {
   }
 
   @override
-  Future<bool> rerunTask(Task task, String idToken) async {
-    assert(idToken != null);
+  Future<CocoonResponse<bool>> rerunTask(Task task, String idToken) async {
+    if (idToken == null || idToken.isEmpty) {
+      return const CocoonResponse<bool>.error('Need to sign in to perform rerun operation');
+    }
 
     final QualifiedTask qualifiedTask = QualifiedTask.fromTask(task);
     assert(qualifiedTask.isLuci);
 
-    /// This endpoint only returns a status code.
+    //this endpoint returns CocoonResponse<bool> which reflects whether the api
+    //call returns success, or returns bad status code.
     final Uri postResetTaskUrl = apiEndpoint('/api/reset-prod-task');
+
     final http.Response response = await _client.post(postResetTaskUrl,
         headers: <String, String>{
           'X-Flutter-IdToken': idToken,
         },
         body: jsonEncode(<String, String>{
           'Key': task.key.child.name,
         }));
-
-    return response.statusCode == HttpStatus.ok;
+    final bool success = response.statusCode == HttpStatus.ok;
+    return success
+        ? const CocoonResponse<bool>.data(true)
+        : CocoonResponse<bool>.error('bad status code is ' +
+            response.statusCode.toString() +
+            ', response body is ' +
+            response.body.toString() +
+            '\n');
```

Comment:
In Dart, you can string interpolation to get a nicer result of:

`HTTP Code: ${response.statusCode}, ${response.body}'`

https://dart.dev/guides/language/language-tour#:~:text=String%20interpolation%3A%20including%20a%20variable%20or%20expression%E2%80%99s%20string%20equivalent%20inside%20of%20a%20string%20literal.%20For%20more%20information%2C%20see%20Strings.